include/zephyr/logging/log_msg2.h

```diff
@@ -276,6 +274,139 @@ do { \
 		(NULL), \
 		(Z_LOG_FMT_ARGS_2(_name, ##__VA_ARGS__)))
 
+#if defined(CONFIG_LOG2_USE_TAGGED_ARGUMENTS)
+
```

Comment:
imo, this should be moved to cbprintf. Right now we have nice separation where cbprintf handles formatted string and now responsibility will be split. Anything preventing from moving it to cbprintf?

Reply:
Moved most of the cbprintf code there.